### PR TITLE
Import code generation tool

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR introduces the `codegen` module which contains the Dart validation code generation tool.

The module was copied from the `base` so that the one in `base` can be removed.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: codegen/build.gradle [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+task copyDartProtobuf(type: Copy) {
+    from protoDart
+}
+
+dependencies {
+    final def protobufDefinitions = [deps.build.protobuf,
+                                     "io.spine:spine-base:$spineBaseVersion",
+                                     "io.spine.tools:spine-tool-base:$spineBaseVersion"]
+    protobuf protobufDefinitions
+    // TODO:2019-10-25:dmytro.dashenkov: Until https://github.com/dart-lang/protobuf/issues/295 is resolved, all types must be compiled in a single batch.
```
Comment (armiol) [inline codegen/build.gradle]: Can you please fit that into 100 symbols? Or at least, 120.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.2-bin.zip
```
Comment (armiol) [inline gradle/wrapper/gradle-wrapper.properties]: Let's migrate to 5.6.3.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM with some comments.

**Pull request merged**